### ISSUE TITLE
Regent: Unexpected behavior with fences and GPU tasks

### ISSUE THREAD

**Issue by sparkasaurusRex** [StanfordLegion/legion#525]
When running the code below on a GPU it immediately prints a super small time but continues to run on the GPU. If I put a fence inside the `launch` task everything behaves as expected. 

```
-- regent bug.rg -ll:gpu 1
import "regent"

__demand(__cuda)
task mycudatask(r_values : region(ispace(int1d), double))
where
  reduces +(r_values)
do
  for e in r_values.ispace do
    for i = 1, 1000000 do
      r_values[e] += i
    end
  end
end


task launch(r_values : region(ispace(int1d), double))
where
  reads writes(r_values)
do
  mycudatask(r_values)
  -- __fence(__execution, __block) -- This fixes the timing issue
end


task toplevel()
  var r_values = region(ispace(int1d, 100000), double)
  fill(r_values, 0)

  __fence(__execution, __block)
  var start_time = regentlib.c.legion_get_current_time_in_micros()
  __fence(__execution, __block)

  -- Should take about 5 seconds
  launch(r_values)

  __fence(__execution, __block)
  var stop_time = regentlib.c.legion_get_current_time_in_micros()
  regentlib.c.printf("Task should have finished now...\n")
  regentlib.c.printf("Runtime: %f\n", (stop_time - start_time) * 1e-6)
end

regentlib.start(toplevel)
```

**Comment by lightsighter** [comment]
Should be fixed in the master branch with 334734da9

**Comment by sparkasaurusRex** [comment]
Looks fixed, thanks!
